src/backends/DomRenderer.js

```diff
@@ -71,7 +72,51 @@ export default class DomRenderer {
   attach() {
     // In the case the element node is external and it is already in the DOM.
     if (this.element.parentNode) return
-    this.head.appendChild(this.element)
+
+    let anchorEl = null
+
+    const {index, jss} = this.options
+    const {registry} = jss.sheets
+
+    if (registry.length > 1) {
+      // Try to insert by index if set
+      if (typeof index === 'number') {
+        for (let i = 0; i < registry.length; i++) {
+          const sheet = registry[i]
+          if (
+            !sheet.attached ||
+            !sheet.options.index ||
+            sheet.options.index < index
+          ) continue
+          anchorEl = sheet.renderer.element
+          break
+        }
+      }
+
+      // Otherwise insert after the last attached
+      if (!anchorEl) {
+        for (let i = registry.length - 1; i >= 0; i--) {
+          const registrySheet = registry[i]
+          if (registrySheet.attached) {
+            anchorEl = registrySheet.renderer.element.nextElementSibling
```

Comment:
@kof it returns `null` which works in `insertBefore` and will just append at the end.